ee/vellum_ee/workflows/display/tests/workflow_serialization/test_integration_trigger_serialization.py

```diff
@@ -212,33 +212,37 @@ class TestWorkflow(BaseWorkflow[BaseInputs, BaseState]):
     assert "source_handle_id" in trigger, "Trigger should have source_handle_id"
     assert trigger["source_handle_id"] == trigger_id, "source_handle_id should match trigger_id"
 
-    # Verify no ENTRYPOINT node exists
+    # Verify ENTRYPOINT node exists
     workflow_raw_data = result["workflow_raw_data"]
     assert isinstance(workflow_raw_data, dict)
     nodes = workflow_raw_data["nodes"]
     assert isinstance(nodes, list)
     entrypoint_nodes = [n for n in nodes if isinstance(n, dict) and n.get("type") == "ENTRYPOINT"]
-    assert len(entrypoint_nodes) == 0, "IntegrationTrigger-only workflows should not have ENTRYPOINT nodes"
+    assert len(entrypoint_nodes) == 1, "IntegrationTrigger workflows should have an ENTRYPOINT node"
 
-    # Verify edges use trigger ID as sourceNodeId
+    entrypoint_node = entrypoint_nodes[0]
+    assert isinstance(entrypoint_node, dict)
+    entrypoint_node_id = entrypoint_node["id"]
+
+    # Verify edges use ENTRYPOINT node ID as sourceNodeId (not trigger ID)
     edges = workflow_raw_data["edges"]
     assert isinstance(edges, list)
-    trigger_edges = [e for e in edges if isinstance(e, dict) and e.get("source_node_id") == trigger_id]
-    assert len(trigger_edges) > 0, "Should have edges from trigger ID"
+    entrypoint_edges = [e for e in edges if isinstance(e, dict) and e.get("source_node_id") == entrypoint_node_id]
+    assert len(entrypoint_edges) > 0, "Should have edges from ENTRYPOINT node"
```

Comment:
DevinAI no, _this_ should still be 0.

Meaning, we should still have an entrypoint node, but the edge between the Trigger and ProcessNode should remain